### PR TITLE
DECO-90 Integrated GNSS in GSM PPP driver

### PR DESCRIPTION
**Pull request by n-nievergeld**
Please change the base before merging.

### CODE REVIEW

File: drivers/modem/gsm_ppp.c [hunk 1]
```diff
@@ -44,6 +47,10 @@ LOG_MODULE_REGISTER(modem_gsm, CONFIG_MODEM_LOG_LEVEL);
 #define GSM_RSSI_RETRIES                10
 #define GSM_RSSI_INVALID                -1000
 
+#if defined(CONFIG_MODEM_GSM_ENABLE_GNSS)
```
Comment (DennisOfTheBurn) [inline drivers/modem/gsm_ppp.c:50]: Why use two different styles? On line 25 you use `#ifdef CONFIG_MODEM_GSM_ENABLE_GNSS`

**Review by DennisOfTheBurn — APPROVED** [review]
Clearly the same implementation as the PoC, noting to note.

**Review by DennisOfTheBurn — COMMENTED** [review]
Sorry, wrong place. I ment to add this comment to the corresponding [DBI](https://github.com/meulemanelectronics/DBI-ICBIS-Economy-System/pull/43#pullrequestreview-1267491321) commit.

File: include/zephyr/drivers/modem/gsm_ppp.h [hunk 2]
```diff
@@ -53,6 +55,41 @@ enum ring_indicator_behaviour {
 	WAVE,
 };
 
+struct gsm_ppp_gnss_data {
```
[the remark this answers is not in the document]
Comment (n-nievergeld) [reply include/zephyr/drivers/modem/gsm_ppp.h:60]: Added ifdef

File: drivers/modem/gsm_ppp.c (hunk 1 above)
Comment (n-nievergeld) [reply drivers/modem/gsm_ppp.c:50]: Changed style to #if defined()

File: include/zephyr/drivers/modem/gsm_ppp.h [hunk 3]
```diff
@@ -23,6 +23,8 @@
 #define GSM_PPP_SMS_DATA_LENGTH         180
 #endif
 
+#define GSM_PPP_GNSS_DATA_UTC_LEN      	 64
```
Comment (DennisOfTheBurn) [inline include/zephyr/drivers/modem/gsm_ppp.h:26]: Maybe add define check here as well.

File: include/zephyr/drivers/modem/gsm_ppp.h [hunk 4]
```diff
@@ -140,4 +179,26 @@ void gsm_ppp_delete_all_sms(const struct device *dev);
 void gsm_ppp_clear_ring_indicator(const struct device *dev);
 #endif /* defined(CONFIG_MODEM_GMS_ENABLE_SMS) */
 
+/**
```
Comment (DennisOfTheBurn) [inline include/zephyr/drivers/modem/gsm_ppp.h:182]: Maybe add define check here as well.

**Pull request merged**